src/Relations/MorphToMany.php

```diff
@@ -0,0 +1,461 @@
+
+        foreach ($results as $result) {
+            if ($this->getInverse()) {
+                foreach ($result->$foreign as $item) {
+                    $dictionary[$item][] = $result;
+                }
+            } else {
+                // Collect $foreign value from pivot column of result model
+                $items = $this->extractIds($result->{$this->table} ?? [], $foreign);
+                foreach ($items as $item) {
+                    $dictionary[$item][] = $result;
+                }
+            }
+        }
+
+        return $dictionary;
+    }
+
+    /** @inheritdoc */
+    public function newPivotQuery()
+    {
+        return $this->newRelatedQuery();
+    }
+
+    /**
+     * Create a new query builder for the related model.
+     *
+     * @return \Illuminate\Database\Query\Builder
+     */
+    public function newRelatedQuery()
+    {
+        return $this->related->newQuery();
+    }
+
+    /**
+     * Get the fully qualified foreign key for the relation.
+     *
+     * @return string
+     */
+    public function getForeignKey()
+    {
+        return $this->foreignPivotKey;
+    }
+
+    /** @inheritdoc */
+    public function getQualifiedForeignPivotKeyName()
+    {
+        return str_contains($this->foreignPivotKey, '.')
+            ? $this->foreignPivotKey
+            : $this->table . '.' . $this->foreignPivotKey;
+    }
+
+    /** @inheritdoc */
+    public function getQualifiedRelatedPivotKeyName()
+    {
+        return $this->relatedPivotKey;
+    }
+
+    /**
+     * Format the sync list so that it is keyed by ID. (Legacy Support)
+     * The original function has been renamed to formatRecordsList since Laravel 5.3.
+     *
+     * @deprecated
+     *
+     * @return array
+     */
+    protected function formatSyncList(array $records)
+    {
+        $results = [];
+        foreach ($records as $id => $attributes) {
+            if (! is_array($attributes)) {
+                [$id, $attributes] = [$attributes, []];
+            }
+
+            $results[$id] = $attributes;
+        }
+
+        return $results;
+    }
+
+    /**
+     * Get the name of the "where in" method for eager loading.
+     *
+     * @param string $key
+     *
+     * @return string
+     */
+    protected function whereInMethod(Model $model, $key)
+    {
+        return 'whereIn';
+    }
+
+    /**
+     * Extract ids from given pivot table data
+     *
+     * @param  array       $data
+     * @param  string|null $relatedPivotKey
+     *
+     * @return mixed
+     */
+    private function extractIds(array $data, ?string $relatedPivotKey = null)
+    {
+        $relatedPivotKey = $relatedPivotKey ?: $this->relatedPivotKey;
+        return array_reduce($data, function ($carry, $item) use ($relatedPivotKey) {
+            if (is_array($item) && array_key_exists($relatedPivotKey, $item)) {
+                $carry[] = $item[$relatedPivotKey];
+            } elseif (is_string($item) && ! str_contains($item, '\\')) {
```

Comment:
This condition is not covered by tests. What is it for?

Reply:
It was related to an issue before I fixed it.